drivers/entropy/entropy_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2019 Brett Witherspoon
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <entropy.h>
+#include <irq.h>
+#include <ring_buffer.h>
+#include <sys_io.h>
+
+#include <driverlib/prcm.h>
+#include <driverlib/trng.h>
+
+struct entropy_cc13xx_cc26xx_data {
+	struct k_sem lock;
+	struct k_sem sync;
+	struct ring_buf pool;
+	u8_t data[CONFIG_ENTROPY_CC13XX_CC26XX_POOL_SIZE];
+};
+
+DEVICE_DECLARE(entropy_cc13xx_cc26xx);
+
+static inline struct entropy_cc13xx_cc26xx_data *
+get_dev_data(struct device *dev)
+{
+	return dev->driver_data;
+}
+
+static int entropy_cc13xx_cc26xx_get_entropy(struct device *dev, u8_t *buf,
+					     u16_t len)
+{
+	struct entropy_cc13xx_cc26xx_data *data = get_dev_data(dev);
+	u32_t cnt;
+
+	TRNGIntEnable(TRNG_NUMBER_READY);
+
+	while (len) {
+		k_sem_take(&data->lock, K_FOREVER);
+		cnt = ring_buf_get(&data->pool, buf, len);
+		k_sem_give(&data->lock);
+
+		if (cnt) {
+			buf += cnt;
+			len -= cnt;
+		} else {
+			k_sem_take(&data->sync, K_FOREVER);
+		}
+	}
+
+	return 0;
+}
+
+static void entropy_cc13xx_cc26xx_isr(void *arg)
+{
+	struct entropy_cc13xx_cc26xx_data *data = get_dev_data(arg);
+	u32_t src, cnt, off;
+	u32_t num[2];
+
+	/* Interrupt service routine as described in TRM section 18.16.1.3.2 */
```

Comment:
I am looking at the TRM at http://www.ti.com/lit/ug/swcu185c/swcu185c.pdf, but there is no section 18.16.1.3.2. Either fix the section number or be more specific about the document referred to.

Reply:
Yes, that is a typo. It should be section 18.6.1.3.2 "Interrupt mode" and in particular figure 18-3 in that section. I will fix.

Reply:
Fixed